src/releaseManager.ts

```diff
@@ -0,0 +1,66 @@
+import * as express from 'express';
+import * as cookieParser from 'cookie-parser';
+import * as httpProxy from 'http-proxy';
+import { first } from 'lodash';
+import { URL } from 'url';
+
+import { health } from './health';
+import { environmentsByUrl } from './routingMap';
+import { createEnvNameGenerator } from './getEnvName';
+import { weightsByEnvironment } from './environments';
+
+const getEnvName = createEnvNameGenerator(weightsByEnvironment);
+
+const proxyServer = httpProxy.createProxyServer();
+
+const server = express();
+
+server.use('/health', health);
+
+// Redirect HTTP requests to HTTPS
+server.use((req, res, next) => {
+  const protocol = req.header('X-FORWARDED-PROTO');
+  const host = req.header('Host') || 'hollowverse.com';
+  if (protocol === 'http') {
+    const newURL = new URL(req.url, `https://${host}`);
+    res.redirect(newURL.toString());
+  } else {
+    next();
+  }
+});
+
+server.use(cookieParser());
+
+server.use(async (req, res) => {
+  const map = await environmentsByUrl;
+  let envName: string | undefined = req.cookies.env;
+  let envUrl: string | undefined;
+  if (!envName || map.get(envName) === undefined) {
+    envName = getEnvName.next().value;
+  }
+
+  // Get the URL from the routing map, falling back to first environment
+  // if the environment is defined but does not have a URL
+  envUrl = map.get(envName);
+  if (!envUrl) {
+    envName = first(Array.from(map.keys())) as string;
```

Comment:
[This StackOverflow](https://stackoverflow.com/a/32373389/604296) has other ways of getting the first element from a JS map.

Reply:
Nice! I changed the code to use the iterator